packages/libp2p/src/autonat/index.ts

```diff
@@ -108,14 +118,16 @@ class DefaultAutoNATService implements Startable {
   private started: boolean
 
   constructor (components: AutoNATComponents, init: AutoNATServiceInit) {
+    const validatedConfig = configValidator.validateSync(init)
```

Comment:
This feels quite verbose and could just be called `config`.
Applies to all places where xxx.validateSync is used.

Reply:
I think `validatedConfig` var name implies that all `undefined` properties of the interface passed in via the constructor will now set to default values,  so I think the name is appropriate, not strongly opinionated but I think the rename has value.

Reply:
> I think validatedConfig var name implies that all undefined properties of the interface passed in via the constructor will now set to default values

I don't think the name implies that. It implies that its a configuration that is validated. Filling in default values is not necessarily a standard step in validation. Validation, imo, speaks to making assertions about unsanitized user inputs.

Moreover, the fact that it is validated or not is not relevant, to the rest of the constructor. We can tell that the config variable is validated because its being assigned to the output of a function called validateSync. We can also tell by its type that its basically a `Required<XXXInit>`. (We could always make that explicit with a type declaration `const config: Required<XXXInit> = xxx.validateSync(init)`)